botocore/credentials.py

```diff
@@ -1321,6 +1323,18 @@ def _validate_source_profile(self, parent_profile, source_profile):
                 )
             )
 
+        profile = profiles[source_profile]
+
+        #ensure the profile has valid credential type
+        if not self._has_static_credentials(profile) and \
+           not self._has_assume_role_config_vars(profile):
+            raise InvalidConfigError(
+                error_msg=(
+                    'The source_profile "%s" must specify either static '
+                    'credentials or a role arn to assume.' % (source_profile)
```

Comment:
I would say "assume role configuration" rather than just mentioning the role arn.

Reply:
Sounds good.